packages/dm-core-plugins/src/job/JobPlugin.tsx

```diff
@@ -188,30 +235,86 @@ export const JobPlugin = (props: IUIPlugin & { config: JobPluginConfig }) => {
         >
           Get results
         </Button>
+        <Button onClick={() => setShowLogs(!showLogs)} variant="ghost">
+          {showLogs ? 'Hide' : 'Show'} logs
+        </Button>
       </JobButtonWrapper>
 
       {status === JobStatus.Failed && (
         <Card variant="danger" style={{ marginTop: '8px' }}>
           <Card.Header>Job status: {status}</Card.Header>
         </Card>
       )}
-      {jobExists && (
-        <>
-          <Typography variant="h6">Logs:</Typography>
-          {error ? (
-            <pre>{JSON.stringify(error, null, 2)}</pre>
-          ) : (
-            <pre>{logs}</pre>
+      {showLogs && (
+        <JobLogBox>
+          <div style={{ position: 'relative' }}>
+            <Button
+              variant="ghost_icon"
+              size={16}
+              label="Expand logs"
+              onClick={() => setShowLogDialog(true)}
+              style={{ position: 'absolute', right: 0, top: -8 }}
+            >
+              <Icon data={expand_screen} />
+            </Button>
+            <div
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                alignItems: 'center',
+                justifyContent: 'space-between',
+                width: '100%',
+              }}
+            >
+              <Typography variant="h6" style={{ paddingBottom: '.4rem' }}>
+                Logs:
+              </Typography>
+              {/*<Button variant="ghost_icon" size={16} label='Expand logs' onClick={() => setShowLogDialog(true)}><Icon*/}
+              {/*  data={expand_screen}/></Button>*/}
+            </div>
+            <FormattedLogContainer
+              style={{ maxWidth: '40rem', maxHeight: '20rem' }}
+            >
+              <code
```

Comment:
Seems to be a bit of duplicated code between this and the "JobsLogsDialog", possible to share some of it?

Reply:
yeah, can probably be extracted to a common place, I just wanted to know about the general direction 👍